src/main/scala/beam/sim/BeamHelper.scala

```diff
@@ -663,6 +693,20 @@ trait BeamHelper extends LazyLogging {
       beamServices.beamConfig
     )
 
+    if (beamServices.beamConfig.beam.agentsim.agents.freight.enabled) {
+      logger.info(s"Generating freight population from ${beamScenario.freightCarriers.size} carriers ...")
+      generatePopulationForPayloadPlans(
+        beamServices.beamConfig,
+        beamServices.geo,
+        beamScenario,
+        scenario.getPopulation,
+        scenario.getHouseholds
+      )
+      logger.info(s"""Freight population generated:
+                     |Number of households: ${scenario.getHouseholds.getHouseholds.keySet.size}
+                     |Number of persons: ${scenario.getPopulation.getPersons.keySet.size}""".stripMargin)
+    }
+
```

Comment:
If we move population generation to runBeam, then `prepareBeamService` will not return these additional persons when is called in other places (i.e. PhyssimReplayer). And we would have inconsistency in this case.

Reply:
I think it makes sense, @nikolayilyin why it was placed there?

Reply:
@dimaopen so this change was made because population generation must be done after population sampling which is located here in `runBeam`. As I checked the other usages of `buildBeamServicesAndScenario` and `prepareBeamService` population from scenario not is not used there, so it should not be the issue.